mobile_packet_verifier/src/daemon.rs

```diff
@@ -0,0 +1,57 @@
+use crate::{burner::Burner, settings::Settings};
+use file_store::{
+    file_info_poller::FileInfoStream, mobile_session::DataTransferSessionIngestReport,
+};
+use sqlx::{Pool, Postgres};
+use tokio::{
+    sync::mpsc::Receiver,
+    time::{sleep_until, Duration, Instant},
+};
+
+pub struct Daemon {
+    pool: Pool<Postgres>,
+    burner: Burner,
+    reports: Receiver<FileInfoStream<DataTransferSessionIngestReport>>,
+    burn_period: Duration,
+}
+
+impl Daemon {
+    pub fn new(
+        settings: &Settings,
+        pool: Pool<Postgres>,
+        reports: Receiver<FileInfoStream<DataTransferSessionIngestReport>>,
+        burner: Burner,
+    ) -> Self {
+        Self {
+            pool,
+            burner,
+            reports,
+            burn_period: Duration::from_secs(60 * 60 * settings.burn_period as u64),
+        }
+    }
+
+    pub async fn run(mut self, shutdown: &triggered::Listener) -> anyhow::Result<()> {
+        let mut burn_time = Instant::now() + self.burn_period;
```

Comment:
maybe a `Settings` method that computes `self.burn_time()` since this seems to always be a static duration determined by the `burn_period` settings value?